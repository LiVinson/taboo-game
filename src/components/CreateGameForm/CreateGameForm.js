import React from 'react'
import PropTypes from 'prop-types'
import { Formik, Field, ErrorMessage } from 'formik'
// import { ErrorMessage as CustomErrorText } from 'components/shared/FeedbackMessage'
import HideContainer from 'components/shared/HideContainer'
import {
	Form,
	FormSection,
	FormSectionTitle,
	InputGroup,
	TextInput,
	RadioButton,
	DropDown,
	Label,
	TextLabel,
	ErrorText,
} from 'components/shared/FormElements'

//Validation occurs onChange or onBlur of inputs and onSubmit of form
const validate = (values) => {
	const errors = {}
	if (!values.name) {
		errors.name = 'Name is Required'
	} else if (values.name.length > 25) {
		errors.name = 'Must be 25 characters or less'
	}
	return errors
}

/*Formik component accepts a function as child component that returns form and props to set defaults.
Function argument is the same object returned by the useFormik hook. Field automatically passes the corresponsing
initial values, onChange, onBlur etc; based on the name of each Field. 
*/

const CreateGameForm = (props) => {
	return (
		<Formik
			initialValues={props.initialValues}
			validate={validate}
			onSubmit={(values, { setSubmitting }) => props.handleSubmit(values, setSubmitting)}
		>
			{(formik) => (
				<Form id="createGameForm">
					<FormSection>
						<TextLabel htmlFor="name">Enter Your Name</TextLabel>
						<Field id="name" name="name" type="text" as={TextInput} placeholder="" tabIndex="1" />
						<ErrorMessage name="name" component={ErrorText} />
					</FormSection>

					{/* LEAVING FOR FUTURE USE: */}

					<HideContainer>
						<FormSection padding="1rem">
							<FormSectionTitle>Game Mode </FormSectionTitle>
							<InputGroup>
								<Field
									type="radio"
									as={RadioButton}
									id="remote"
									name="gameMode"
									value="remote"
									tabIndex="2"
								/>
								<Label htmlFor="remote">Remote</Label>
							</InputGroup>
							<InputGroup>
								<Field type="radio" as={RadioButton} id="inPerson" name="gameMode" value="inPerson" />
								<Label htmlFor="inPerson">In Person</Label>
							</InputGroup>
						</FormSection>
          </HideContainer>
          

					{/* Radio button and drop down combination. Drop down for # of turns/amount of time is disabled if 
                corresponding radio button for that option is not selected */}

					<FormSection padding="1rem">
						<FormSectionTitle>End Game Method</FormSectionTitle>
						<InputGroup>
							<Field
								type="radio"
								as={RadioButton}
								id="turns"
								name="endGameMethod"
								value="turns"
								tabIndex="3"
							/>
							<Label htmlFor="turns">Turns per Person</Label>
						</InputGroup>
						<Field
							name="turnsValue"
							as={DropDown}
							disabled={formik.values.endGameMethod === 'time'}
							tabIndex="4"
						>
							<option value={1}>1 Turn</option>
							<option value={2}>2 Turns</option>
							<option value={3}>3 Turns</option>
							<option value={4}>4 Turns</option>
							<option value={5}>5 Turns</option>
            </Field>
            {/* LEAVING FOR FUTURE USE: */}
						<HideContainer>
							<InputGroup>
								<Field
									as={RadioButton}
									type="radio"
									id="time"
									name="endGameMethod"
									value="time"
									tabIndex="5"
									disabled={true}
								/>
								<Label htmlFor="time">Timer (Coming soon!)</Label>
							</InputGroup>
							<Field
								name="timeValue"
								as={DropDown}
								// disabled={formik.values.endGameMethod === "turns"}
								disabled={true}
								tabIndex="6"
							>
								<option value={30}>30 minutes</option>
								<option value={60}>60 minutes</option>
								<option value={90}>90 minutes</option>
							</Field>
						</HideContainer>
					</FormSection>

					<FormSection padding="1rem">
						<FormSectionTitle>Skipping Penalty</FormSectionTitle>
						<InputGroup>
							<Field
								as={RadioButton}
								type="radio"
								id="none"
								name="skipPenalty"
								value="none"
								tabIndex="7"
							/>
							<Label htmlFor="none">None</Label>
						</InputGroup>
						<InputGroup>
							<Field as={RadioButton} type="radio" id="half" name="skipPenalty" value="half" />
							<Label htmlFor="half">1/2 a Point</Label>
						</InputGroup>
						<InputGroup>
							<Field as={RadioButton} type="radio" id="full" name="skipPenalty" value="full" />
							<Label htmlFor="full">1 Point</Label>
						</InputGroup>
					</FormSection>
					{/* Submit Button is passed down to Taboo Card for UI consistency reasons*/}
				</Form>
			)}
		</Formik>
	)
}

CreateGameForm.propTypes = {
  initialValues: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default CreateGameForm
