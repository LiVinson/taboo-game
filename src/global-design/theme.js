const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
