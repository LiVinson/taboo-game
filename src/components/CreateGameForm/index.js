import CreateGameForm from "./CreateGameForm"

export default CreateGameForm