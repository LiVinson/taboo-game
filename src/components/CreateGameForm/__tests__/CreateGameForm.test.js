import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import theme from "../../../global-design/theme"
import CreateGameForm from "components/CreateGameForm"

test("CreateGameForm renders correctly", () => {
  const props = {
    handleSubmit: jest.fn(),
    initialValues: {
      name: "",
      gameMode: "remote",
      endGameMethod: "turns",
      turnsValue: 2,
      timeValue: 60,
      skipPenalty: "none",
    },
  }

  const wrapper = renderer.create(
    <ThemeProvider theme={theme}>
      <CreateGameForm {...props} />
    </ThemeProvider>
  )
  expect(wrapper).toMatchSnapshot()
})


