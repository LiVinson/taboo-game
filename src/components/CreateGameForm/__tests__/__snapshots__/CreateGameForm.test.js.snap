// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGameForm renders correctly 1`] = `
<form
  action="#"
  className="sc-AxirZ jZLUev"
  id="createGameForm"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="sc-AxiKw ebFOUE"
  >
    <label
      className="sc-AxmLO sc-fzozJi kZdFWe"
      htmlFor="name"
    >
      Enter Your Name
    </label>
    <input
      className="sc-AxgMl egcgFI"
      id="name"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="1"
      type="text"
      value=""
    />
  </div>
  <span
    className="sc-AxjAm htLRco"
  >
    <div
      className="sc-AxiKw dsnoEE"
    >
      <p
        className="sc-AxhCb bDNQHH"
      >
        Game Mode 
      </p>
      <div
        className="sc-AxhUy fehFXc"
      >
        <input
          checked={true}
          className="sc-AxheI jnBKJW"
          id="remote"
          name="gameMode"
          onBlur={[Function]}
          onChange={[Function]}
          tabIndex="2"
          type="radio"
          value="remote"
        />
        <label
          className="sc-AxmLO XqAik"
          htmlFor="remote"
        >
          Remote
        </label>
      </div>
      <div
        className="sc-AxhUy fehFXc"
      >
        <input
          checked={false}
          className="sc-AxheI jnBKJW"
          id="inPerson"
          name="gameMode"
          onBlur={[Function]}
          onChange={[Function]}
          type="radio"
          value="inPerson"
        />
        <label
          className="sc-AxmLO XqAik"
          htmlFor="inPerson"
        >
          In Person
        </label>
      </div>
    </div>
  </span>
  <div
    className="sc-AxiKw dsnoEE"
  >
    <p
      className="sc-AxhCb bDNQHH"
    >
      End Game Method
    </p>
    <div
      className="sc-AxhUy fehFXc"
    >
      <input
        checked={true}
        className="sc-AxheI jnBKJW"
        id="turns"
        name="endGameMethod"
        onBlur={[Function]}
        onChange={[Function]}
        tabIndex="3"
        type="radio"
        value="turns"
      />
      <label
        className="sc-AxmLO XqAik"
        htmlFor="turns"
      >
        Turns per Person
      </label>
    </div>
    <select
      className="sc-Axmtr hzlKJe"
      disabled={false}
      name="turnsValue"
      onBlur={[Function]}
      onChange={[Function]}
      tabIndex="4"
      value={2}
    >
      <option
        value={1}
      >
        1 Turn
      </option>
      <option
        value={2}
      >
        2 Turns
      </option>
      <option
        value={3}
      >
        3 Turns
      </option>
      <option
        value={4}
      >
        4 Turns
      </option>
      <option
        value={5}
      >
        5 Turns
      </option>
    </select>
    <span
      className="sc-AxjAm htLRco"
    >
      <div
        className="sc-AxhUy fehFXc"
      >
        <input
          checked={false}
          className="sc-AxheI jnBKJW"
          disabled={true}
          id="time"
          name="endGameMethod"
          onBlur={[Function]}
          onChange={[Function]}
          tabIndex="5"
          type="radio"
          value="time"
        />
        <label
          className="sc-AxmLO XqAik"
          htmlFor="time"
        >
          Timer (Coming soon!)
        </label>
      </div>
      <select
        className="sc-Axmtr hzlKJe"
        disabled={true}
        name="timeValue"
        onBlur={[Function]}
        onChange={[Function]}
        tabIndex="6"
        value={60}
      >
        <option
          value={30}
        >
          30 minutes
        </option>
        <option
          value={60}
        >
          60 minutes
        </option>
        <option
          value={90}
        >
          90 minutes
        </option>
      </select>
    </span>
  </div>
  <div
    className="sc-AxiKw dsnoEE"
  >
    <p
      className="sc-AxhCb bDNQHH"
    >
      Skipping Penalty
    </p>
    <div
      className="sc-AxhUy fehFXc"
    >
      <input
        checked={true}
        className="sc-AxheI jnBKJW"
        id="none"
        name="skipPenalty"
        onBlur={[Function]}
        onChange={[Function]}
        tabIndex="7"
        type="radio"
        value="none"
      />
      <label
        className="sc-AxmLO XqAik"
        htmlFor="none"
      >
        None
      </label>
    </div>
    <div
      className="sc-AxhUy fehFXc"
    >
      <input
        checked={false}
        className="sc-AxheI jnBKJW"
        id="half"
        name="skipPenalty"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="half"
      />
      <label
        className="sc-AxmLO XqAik"
        htmlFor="half"
      >
        1/2 a Point
      </label>
    </div>
    <div
      className="sc-AxhUy fehFXc"
    >
      <input
        checked={false}
        className="sc-AxheI jnBKJW"
        id="full"
        name="skipPenalty"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="full"
      />
      <label
        className="sc-AxmLO XqAik"
        htmlFor="full"
      >
        1 Point
      </label>
    </div>
  </div>
</form>
`;
